pkg/controller/acmeorders/sync.go

```diff
@@ -71,11 +71,22 @@ func (c *Controller) Sync(ctx context.Context, o *cmapi.Order) (err error) {
 
 	if o.Status.URL == "" {
 		err := c.createOrder(ctx, cl, genericIssuer, o)
-		// TODO: check for error types (perm or transient?)
+
 		if err != nil {
-			// TODO: check for acmeerrors.IsMalformed and mark the Order as failed
+			// If we get a 4xx error, we mark the Order as 'error'.
+			// 4xx error codes include rate limit errors (429).
+			// This will cause the Certificate controller to retry the Order
+			// after the regular back-off algorithm has been applied.
+			acmeErr, ok := err.(*acmeapi.Error)
+			if ok && acmeErr.StatusCode >= 400 && acmeErr.StatusCode < 500 {
+				c.setOrderState(&o.Status, cmapi.Errored)
+				o.Status.Reason = fmt.Sprintf("Failed to create order: %v", err)
+				return err
```

Comment:
nit: this return isn't really needed

Reply:
Will follow up as there's some other changes to this code area coming up 😄